File: Baekjoon/BaaarkingDog/0x03_array/1158.cpp
// 1158: 요세푸스 문제

#include <iostream>
#include <queue>
using namespace std;

int main() {
  int N, K, cnt = 1;
  cin >> N >> K;
  
  queue<int> q;

  for (int i=0; i<N; i++) q.push(i+1);

  cout << '<';

  while (q.size() != 1) {
    int front = q.front();
    q.pop();
    if (cnt++ % K != 0) q.push(front);
    else cout << front << ", ";
  }

  cout << q.front() << '>';

  return 0;
}

// <-- 1 2 3 4 5 6 7 <--

File: Baekjoon/BaaarkingDog/0x03_array/1158.py
# 1158: 요세푸스 문제

from collections import deque

N, K = list(map(int, input().split()))

q = deque(range(1, N+1)) # [i+1 for i in range(N)]

print('<', end="")

cnt = 1
while (len(q) != 1):
  if (cnt % K != 0): q.append(q.popleft()) # 기존 코드: del q[0], append() 활용 -> O(N)
  else: print(f"{q.popleft()}, ", end="")

  cnt += 1

print(f"{q.pop()}>")
